entity-framework/core/get-started/winforms.md

```diff
@@ -0,0 +1,318 @@
+
+The `OnClosing` method is called when the form is closed. At this time, the `DbContext` is disposed, which ensures any database resources will be freed, and the `dbContext` field is set to null so that it cannot be used again.
+
+## Populating the Products view
+
+If the application is started at this point, then it should look something like this:
+
+![Fist run of the application](_static/winforms-first-run.png)
+
+Notice that the categories have been loaded from the database, but the products table remains empty. Also, the **Save** button does not work.
+
+To populate the products table, EF Core needs to load products from the database for the selected category. To achieve this:
+
+1. In the designer for the main form, select the `DataGridView` for categories.
+2. In the **Properties** for the `DataGridView`, choose the events (the lightning button), and double click the **SelectionChanged** event.
+
+   ![Add the SelectionChanged event](_static/winforms-selection-changed-event.png)
+
+   This will create stub in the main form code for an event to be fired whenever the category selection changes.  
+
+3. Fill in the code for the event:
+
+```csharp
+private void dataGridViewCategories_SelectionChanged(object sender, EventArgs e)
+{
+    if (this.dbContext != null)
+    {
+        var category = (Category)this.dataGridViewCategories.CurrentRow.DataBoundItem;
+
+        if (category != null)
+        {
+            this.dbContext.Entry(category).Collection(e => e.Products).Load();
+        }
+    }
+}
+```
+
+In this code, if there is an active (non-null) `DbContext` session, then we obtain the `Category` instance bound to the currently selected row of the `DataViewGrid`. (This may be `null` if the final row in the view is selected, which is used to create new categories.) If there is a selected category, then the `DbContext` is instructed to load the products associated with that category. This is done by:
+
+- Getting an `EntityEntry` for the `Category` instance (`dbContext.Entry(category)`)
+- Letting EF Core know that we want to operate on the `Products` collection navigation of that `Category` (`.Collection(e => e.Products)`)
+- And finally telling EF Core that we want to load that collection of products from the database (`.Load();`)
+
+> [!TIP]
+> When `Load` is called, EF Core will only access the database to load the products if they have not already been loaded.
+
+If the application is now run again, then it should load the appropriate products whenever a category is selected:
+
+![Products are loaded](_static/winforms-products-are-loaded.png)
+
+## Saving changes
+
+Finally, the **Save** button con be connected to EF Core so that any changes made to the products and categories are saved to the database.
```

Comment:
```suggestion
Finally, the **Save** button can be connected to EF Core so that any changes made to the products and categories are saved to the database.
```